services/getstream/followTopic.js

```diff
@@ -49,24 +49,56 @@ const addTopicToChatTab = async (token, topicName, userId, isAnonymous) => {
     const text = isAnonymous
       ? 'This topic has new anonymous followers'
       : 'This topic has new followers';
-    const defaultImage = isAnonymous ? DEFAULT_TOPIC_PIC_PATH_ANON : DEFAULT_TOPIC_PIC_PATH_SIGN;
+
+    const admin = new StreamChat(Environment.GETSTREAM_API_KEY, Environment.GETSTREAM_API_SECRET);
+    const adminChannel = admin.channel('topics', `topic_${topic}`);
+    const adminChannelData = await adminChannel.create();
+
+    const updateData = {};
+    // Revert back all existing signed channel that has anon topic to correct signed one
+    if (
+      [DEFAULT_TOPIC_PIC_PATH_ANON, DEFAULT_TOPIC_PIC_PATH_SIGN].includes(
+        adminChannelData?.channel?.image
+      )
+    ) {
+      updateData.image = null;
+    }
+
+    if (
+      [DEFAULT_TOPIC_PIC_PATH_ANON, DEFAULT_TOPIC_PIC_PATH_SIGN].includes(
+        adminChannelData?.channel?.channelImage
+      )
+    ) {
+      updateData.channelImage = null;
+    }
+
+    if (
+      [DEFAULT_TOPIC_PIC_PATH_ANON, DEFAULT_TOPIC_PIC_PATH_SIGN].includes(
+        adminChannelData?.channel?.channel_image
+      )
+    ) {
+      updateData.channel_image = null;
+    }
+    // Revert END
+
+    await adminChannel.updatePartial({
+      set: updateData
+    });
 
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The code creates a new instance of `StreamChat` as `admin` and another one as `client`. This seems unnecessary and could potentially lead to performance issues due to the creation of multiple instances. Consider reusing the same instance for both admin and client operations.

```diff
-    const admin = new StreamChat(Environment.GETSTREAM_API_KEY, Environment.GETSTREAM_API_SECRET);
+    const client = new StreamChat(Environment.GETSTREAM_API_KEY, Environment.GETSTREAM_API_SECRET);
-    const adminChannel = admin.channel('topics', `topic_${topic}`);
+    const adminChannel = client.channel('topics', `topic_${topic}`);
     const adminChannelData = await adminChannel.create();
     
     // ... rest of the code

-    const client = new StreamChat(Environment.GETSTREAM_API_KEY, Environment.GETSTREAM_API_SECRET);
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->